### PR TITLE
[Merged by Bors] - feat: convergent series in the completion of a group

### PR DESCRIPTION
**Pull request by trivial1711**
Prove the following.
- `Inducing.hasProd_iff` / `Inducing.hasSum_iff`: Let $g \colon \beta \to \alpha$ be an inducing homomorphism of topological additive commutative monoids. A series $\sum_{i \in I} b_i$ unconditionally converges to $b \in \beta$ if and only if $\sum_{i \in I} g(b_i)$ unconditionally converges to $g(b)$. (That is, if $g$ is inducing, then the converse of `HasProd.map` / `HasSum.map` holds.)
- `Multipliable.map_tprod` / `Summable.map_tsum`: Let $g \colon \beta \to \alpha$ be a continuous homomorphism of topological additive commutative monoids. If the series $\sum_{i \in I} b_i$ converges unconditionally, then $g\left(\sum_{i \in I} b_i\right) = \sum_{i \in I} g(b_i)$.
- `Inducing.multipliable_iff_tprod_comp_mem_range` / `Inducing.summable_iff_tsum_comp_mem_range`: Let $g \colon \beta \to \alpha$ be an inducing homomorphism of topological additive commutative monoids. Assume that $\alpha$ is Hausdorff. Then a series $\sum_{i \in I} b_i$ converges unconditionally if and only if the series $\sum_{i \in I} g(b_i)$ converges unconditionally and its value lies in the range of $g$.
- `hasSum_iff_hasSum_compl`: Let $\alpha$ be a uniform additive abelian group. An equation $\sum_{i \in I} a_i = a$ holds in $\alpha$ if and only if it holds in $\hat{\alpha}$.
- `summable_iff_summable_compl_and_tsum_mem`:  Let $\alpha$ be a uniform additive abelian group. A series $\sum_{i \in I} a_i$ is unconditionally summable in $\alpha$ if and only if it is unconditionally summable in $\hat{\alpha}$ and its value is in $\alpha$.
- `summable_iff_cauchySeq_finset_and_tsum_mem`: Let $\alpha$ be a uniform additive abelian group. A series $\sum_{i \in I} a_i$ is unconditionally summable in $\alpha$ if and only if its partial sums form a Cauchy net and its value in $\hat{\alpha}$ lies in $\alpha$.
- `Summable.toCompl_tsum`: If a series $\sum_{i \in I} a_i$ is unconditionally summable in $\alpha$, then its value in $\alpha$ is equal to its value in $\hat{\alpha}$.

Note: This PR does not include multiplicative analogues of `hasSum_iff_hasSum_compl`, `summable_iff_summable_compl_and_tsum_mem`, `summable_iff_cauchySeq_finset_and_tsum_mem`, or `Summable.toCompl_tsum`. This is because mathlib does not currently put a group structure on the completion of a uniform multiplicative group (see #11837). Once the group structure on the completion of a uniform multiplicative group is defined, the multiplicative analogues of these four theorems can easily be added.

Note: The argument `(g : G)` of `Multipliable.map_tprod` is explicit to match `Multipliable.map`. At some point, it might be desirable to make these arguments (and several others) implicit, but that is outside the scope of this PR.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/Algebra/InfiniteSum/Basic.lean [hunk 1]
```diff
@@ -220,6 +227,27 @@ protected theorem Multipliable.map_iff_of_leftInverse [CommMonoid γ] [Topologic
...
 #align summable.map_iff_of_left_inverse Summable.map_iff_of_leftInverse
 
+@[to_additive]
+theorem Multipliable.map_tprod [CommMonoid γ] [TopologicalSpace γ] [T2Space γ] (hf : Multipliable f)
+    {G} [FunLike G α γ] [MonoidHomClass G α γ] (g : G) (hg : Continuous g) :
+    g (∏' i, f i) = ∏' i, g (f i) := (HasProd.tprod_eq (HasProd.map hf.hasProd g hg)).symm
+
+@[to_additive]
+theorem Inducing.multipliable_iff_tprod_comp_mem_range [CommMonoid γ] [TopologicalSpace γ]
+    [T2Space γ] {G} [FunLike G α γ] [MonoidHomClass G α γ] {g : G} (hg : Inducing g) (f : β → α):
```
Comment (fpvandoorn) [inline Mathlib/Topology/Algebra/InfiniteSum/Basic.lean:237]:
```suggestion
    [T2Space γ] {G} [FunLike G α γ] [MonoidHomClass G α γ] {g : G} (hg : Inducing g) (f : β → α) :
```

Comment (fpvandoorn) [reply Mathlib/Topology/Algebra/InfiniteSum/Basic.lean:237]: please also edit other occurrences of this below

File: Mathlib/Topology/Algebra/InfiniteSum/Group.lean [hunk 2]
```diff
@@ -318,6 +319,49 @@ theorem prod_mul_tprod_subtype_compl [T2Space α] {f : β → α} (hf : Multipli
...
+section UniformAddGroup
+
+open UniformSpace.Completion
+
+variable [AddCommGroup α] [UniformSpace α] [UniformAddGroup α]
+
+/-- A function `f` has a sum in an uniform additive group `α` if and only if it has that sum in the
+completion of `α`. -/
+theorem hasSum_iff_hasSum_compl (f : β → α) (a : α):
+    HasSum f a ↔ HasSum (toCompl ∘ f) a := (denseInducing_toCompl α).hasSum_iff f a
```
Comment (fpvandoorn) [inline Mathlib/Topology/Algebra/InfiniteSum/Group.lean:331]: also reverse iff.

File: Mathlib/Topology/Algebra/InfiniteSum/Group.lean [hunk 3]
```diff
@@ -318,6 +319,49 @@ theorem prod_mul_tprod_subtype_compl [T2Space α] {f : β → α} (hf : Multipli
...
+    apply (summable_iff_summable_compl_and_tsum_mem f).mpr
+    constructor
+    · apply summable_iff_cauchySeq_finset.mpr
+      simp_rw [comp_apply, ← map_sum]
+      exact h_cauchy.map (uniformContinuous_coe α)
+    · exact h_tsum
+
+/-- If a function `f` is summable in a uniform additive group `α`, then its sum in `α` is the same
+as its sum in `Completion α`. -/
+theorem Summable.toCompl_tsum {f : β → α} (hf : Summable f) : ∑' i, f i = ∑' i, toCompl (f i) :=
```
Comment (fpvandoorn) [inline Mathlib/Topology/Algebra/InfiniteSum/Group.lean:360]: reverse equality

File: Mathlib/Topology/Algebra/InfiniteSum/Group.lean [hunk 4]
```diff
@@ -1,10 +1,11 @@
 /-
 Copyright (c) 2017 Johannes Hölzl. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
-Authors: Johannes Hölzl
+Authors: Johannes Hölzl, Mitchell Lee
 -/
 import Mathlib.Topology.Algebra.InfiniteSum.Basic
 import Mathlib.Topology.Algebra.UniformGroup
+import Mathlib.Topology.Algebra.GroupCompletion
```
Comment (fpvandoorn) [inline Mathlib/Topology/Algebra/InfiniteSum/Group.lean:8]: Please move the new results to a new file (`InfiniteSum/GroupCompletion`?) unless this transitively imports very few new files.

**Review by fpvandoorn — COMMENTED** [review]
Sorry for the late review. Looks good, but here are some comments.

**Comment by trivial1711** [comment]
Thank you! I have addressed your comments and I think this is ready to be reviewed again.

File: Mathlib/Topology/Algebra/InfiniteSum/GroupCompletion.lean [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+# Infinite sums in the completion of a topological group
+-/
+
+open UniformSpace.Completion
+
+variable {α β : Type*} [AddCommGroup α] [UniformSpace α] [UniformAddGroup α]
+
+/-- A function `f` has a sum in an uniform additive group `α` if and only if it has that sum in the
+completion of `α`. -/
+theorem hasSum_iff_hasSum_compl (f : β → α) (a : α):
```
Comment (fpvandoorn) [inline Mathlib/Topology/Algebra/InfiniteSum/GroupCompletion.lean:19]:
```suggestion
theorem hasSum_iff_hasSum_compl (f : β → α) (a : α) :
```

**Review by fpvandoorn — COMMENTED** [review]
Thanks

bors d+

**Comment by fpvandoorn** [comment]
This is not a big problem, but in this case I allowed you to merge it, but I also wrote another comment with a missing space. The intention was of course that you first wrote a commit adding that space, and merged it only afterwards.

**Pull request closed without merging**